controller/destination/server.go

```diff
@@ -286,6 +291,15 @@ func splitDNSName(dnsName string) ([]string, error) {
 		if l == "" {
 			return []string{}, errors.New("Empty label in DNS name: " + dnsName)
 		}
+		if !dnsCharactersRegexp.MatchString(l) {
+			return []string{}, errors.New("DNS name is too long or contains invalid characters: " + dnsName)
+		}
+		if !startsAndEndsWithAlphanumRegexp.MatchString(l) {
```

Comment:
I'd just check that the first and last characters aren't equal to `-`, which will have the same effect without rescanning the entire input string again, and clearer.

(I think "^[a-zA-Z0-9]([a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?$" can be used to combine the two REs above, but I don't think that's clearer than just checking that the first and last characters aren't "-".)